File: web/src/main/java/com/github/keraton/triptr/note/util/NoteUtils.java
package com.github.keraton.triptr.note.util;

import com.github.keraton.triptr.domain.model.NoteCategoryTrip;
import org.springframework.stereotype.Component;

import static com.github.keraton.triptr.domain.model.NoteCategoryTrip.*;

@Component
public class NoteUtils {

    public NoteCategoryTrip calculateHotelNote(Integer stars) {
        switch (stars) {
            case 5:
                return EXCEPTIONAL;
            case 4:
                return GOOD;
            case 3:
                return ACCEPTABLE;
            case 2:
                return ACCEPTABLE;
            default:
                return BAD;
        }
    }

    public NoteCategoryTrip calculateRestaurantNote(Integer stars) {
        switch (stars) {
            case 3:
                return EXCEPTIONAL;
            case 2:
                return GOOD;
            case 1:
                return GOOD;
            default:
                return BAD;
        }
    }
}

File: note/src/main/java/com/github/keraton/triptr/note/util/NoteUtils.kt
package com.github.keraton.triptr.note.util

import com.github.keraton.triptr.domain.model.NoteCategoryTrip
import com.github.keraton.triptr.domain.model.NoteCategoryTrip.*
import org.springframework.stereotype.Component

@Component
class NoteUtils {

    fun calculateHotelNote(stars: Int?): NoteCategoryTrip {
        when (stars) {
            5 -> return EXCEPTIONAL
            4 -> return GOOD
            3 -> return ACCEPTABLE
            2 -> return ACCEPTABLE
            else -> return BAD
        }
    }

    fun calculateRestaurantNote(stars: Int?): NoteCategoryTrip {
        when (stars) {
            3 -> return EXCEPTIONAL
            2 -> return GOOD
            1 -> return GOOD
            else -> return BAD
        }
    }
}
